### ISSUE TITLE
synthseg slim

### ISSUE THREAD

**Issue by jordandekraker** [khanlab/synthseg_nnunet#1]
SynthSeg has a lot of pinned dependencies and is not really actively maintained, os i think that instead of trying to implement it fully in a separate conda environment, we should simply import the specific functionalities we need (and update to accomodate dependency updates too)

The main call we currently need from synthseg is [brain_generator](https://github.com/BBillot/SynthSeg/blob/master/SynthSeg/brain_generator.py). This still calls a few other modules, but shouldn't be too bad since we won't need most of the stuff in that repo. 

The second thing we'll need to do is adapt nnunet's image loader to work with single image-label pairs that are in memory rather than a directory of image-label pairs. There's not much documentation on the code, so this may bea bit tricky. Some ways we could likely simplify this are setting the batch size to 1 (which is not a bad idea anyway since we will likely want to maximize patch size in limited gpu memory). 

I don't think its realistic to have brain_generator operate on gpu, which would be fast but would require significant work. instead, i expect we'll pass single image-label pairs to gpu and train from there. 

my original workflow was:
load labelmap to memory > brain_generator > save label-image pairs to disk (x10k) > preprocess label-image pairs metadata >  load single pairs to nnunet memory > augment pairs > gpu memory > train
now we should have something like:
load labelmap to memory > brain_generator > predefined metadata > pytorch_nnunet loads to gpu, augments, trains on single image pairs
